internal/core/textlayout/sharedparley.rs

```diff
@@ -0,0 +1,138 @@
+// Copyright © SixtyFPS GmbH <info@slint.dev>
+// SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-Royalty-free-2.0 OR LicenseRef-Slint-Software-3.0
+
+pub use parley;
+
+use crate::{
+    graphics::FontRequest,
+    items::TextStrokeStyle,
+    lengths::LogicalLength,
+    textlayout::{TextHorizontalAlignment, TextOverflow, TextVerticalAlignment, TextWrap},
+};
+use i_slint_common::sharedfontique;
+
+pub const DEFAULT_FONT_SIZE: LogicalLength = LogicalLength::new(12.);
+
+fn font_context() -> parley::FontContext {
+    parley::FontContext {
+        collection: sharedfontique::COLLECTION.inner.clone(),
+        source_cache: sharedfontique::COLLECTION.source_cache.clone(),
+    }
+}
+
+static LAYOUT_CONTEXT: std::sync::LazyLock<parley::LayoutContext<Brush>> =
+    std::sync::LazyLock::new(|| Default::default());
+
+#[derive(Debug, Default, PartialEq, Clone, Copy)]
+pub struct Brush {
+    pub is_selected: bool,
+    pub stroke: Option<TextStrokeStyle>,
+}
+
+pub struct LayoutOptions {
+    pub max_width: Option<LogicalLength>,
+    pub max_height: Option<LogicalLength>,
+    pub horizontal_align: TextHorizontalAlignment,
+    pub vertical_align: TextVerticalAlignment,
+    pub stroke: Option<TextStrokeStyle>,
+    pub selection: Option<std::ops::Range<usize>>,
+    pub font_request: Option<FontRequest>,
+    pub text_wrap: TextWrap,
+    pub text_overflow: TextOverflow,
+}
+
+impl Default for LayoutOptions {
+    fn default() -> Self {
+        Self {
+            max_width: None,
+            max_height: None,
+            horizontal_align: TextHorizontalAlignment::Left,
+            vertical_align: TextVerticalAlignment::Top,
+            stroke: None,
+            selection: None,
+            font_request: None,
+            text_wrap: TextWrap::WordWrap,
+            text_overflow: TextOverflow::Clip,
+        }
+    }
+}
+
+pub fn layout(text: &str, scale_factor: f32, options: LayoutOptions) -> Layout {
+    let mut font_context = font_context();
+    let mut layout_context = LAYOUT_CONTEXT.clone();
```

Comment:
To me this seems identical to creating a new layout context. Did you mean to share this? (I think it could go into a thread_local refcell)

Reply:
Ah yep I think this has to be thread local actually.